### PR TITLE
Save screenshots as PNG

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: src/openloco/input/keyboard.cpp [hunk 1]
```diff
@@ -417,23 +420,97 @@ namespace openloco::input
...
+        auto ostream = static_cast<std::ostream*>(png_get_io_ptr(png_ptr));
+        ostream->flush();
+    }
+
     // 0x00452667
     static int16_t saveScreenshot()
     {
-        registers regs;
+        void* buffer = malloc(0x10000);
+        if (buffer == nullptr)
```
Comment (AaronVanGeffen) [inline src/openloco/input/keyboard.cpp]: This buffer is unused, right?

Comment (marijnvdwerf) [reply src/openloco/input/keyboard.cpp]: Probably just a remnant of the BMP decompilation I thought of decompiling before.

**Comment by AaronVanGeffen** [comment]
Depends on #362 and is pending a changelog entry, but other than that: LGTM.

**Comment by IntelOrca** [comment]
The warning is genunine, you shouldn't mix C++ exceptions with `setjmp`.

**Comment by AaronVanGeffen** [comment]
@IntelOrca Ah, interesting. Given that's it's literally what the libpng docs say, I assumed it was fine. How would you change the code to prevent the warning?

**Comment by IntelOrca** [comment]
Do you have a link to the docs? Maybe some compilers can do it. Anyway, it is a C library so you will have to do it the C way. Duplicate dispose calls on all tails, a goto end label or micro routines that return exit codes appropriately.

I always found setjmp a dodgey solution to error handling over exit codes anyway, I don't know if you can avoid setjmp in libpng or not.

**Comment by AaronVanGeffen** [comment]
I'm not seeing any errors on macOS or Linux, no, but it could just be that they've not been enabled.

The docs are over at http://libpng.org/pub/png/libpng-manual.txt

**Comment by janisozaur** [comment]
Libpng is infamous for the setjmp stuff. But we can still avoid it:

> 
If you would rather avoid the complexity of setjmp/longjmp issues,
you can compile libpng with PNG_NO_SETJMP, in which case
errors will result in a call to PNG_ABORT() which defaults to abort().

> You can #define PNG_ABORT() to a function that does something
more useful than abort(), as long as your function does not
return.

**Comment by IntelOrca** [comment]
> in which case errors will result in a call to PNG_ABORT() which defaults to abort()

That isn't anymore helpful, we want to handle the errors gracefully.

**Comment by IntelOrca** [comment]
On Linux it will load the shared version of libpng which won't behave how you want though right?

**Comment by AaronVanGeffen** [comment]
Thanks for the feedback I hope I have come up with an acceptable solution in 3cd1761050e8edf02d3792d9936a4e020f336920 ?

**Comment by IntelOrca** [comment]
> Any thoughts? Do we really need to avoid setjmp entirely? If so, how does OpenRCT2 achieve this?

No you can use it, just isolate in a function without any C++ exception throws or catches.

**Comment by AaronVanGeffen** [comment]
It seems the Ubuntu Docker images were never updated to include libpng either, despite the Dockerfiles having been updated. The cause seems to be a broken webhook. We still need a solution for MinGW too.

@janisozaur Could you help out with this? I've posted some details in the team chat.

File: data/language/de-DE.yml [hunk 2]
```diff
@@ -118,7 +118,7 @@ strings:
   104: Spiel speichern
   105: Szenario-Editor beenden
   106: Spiel beenden
-  107: Screenshot{INT16}.BMP
+  107: Screenshot{INT16}.png
```
Comment (marijnvdwerf) [inline data/language/de-DE.yml]: Just put no file extension in the translation files. We'll always save as png anyways, and if we ever add something like webp-support, we don't need to change the translations.

**Pull request merged**